### PR TITLE
[ISSUE-134] Consolidate CSV and FastCSV Suites

### PR DESCRIPTION
**Pull request by brkyvz**
This will make sure that the exact same test runs for both suites with minimal effort.

### CODE REVIEW

File: src/test/scala/com/databricks/spark/csv/CsvSuite.scala [hunk 1]
```diff
@@ -156,17 +165,26 @@ class CsvSuite extends FunSuite {
...
+        parserLib = parserLib)
+        .select("year")
+        .collect()
 
     assert(results.size === numCars)
   }
 
-
-  test("Expect parsing error with wrong delimiter setting using sparkContext.csvFile") {
+  test("Expect parsing error with wrong delimiter settting using sparkContext.csvFile") {
```
Comment (falaki) [inline src/test/scala/com/databricks/spark/csv/CsvSuite.scala]: nit: setting

File: src/test/scala/com/databricks/spark/csv/CsvSuite.scala [hunk 2]
```diff
@@ -342,20 +359,16 @@ class CsvSuite extends FunSuite {
...
 
-
   test("DSL test schema inferred correctly") {
-
-    val results = new CsvParser()
-      .withInferSchema(true)
-      .withUseHeader(true)
-      .csvFile(TestSQLContext, carsFile)
+    val results = TestSQLContext
+      .csvFile(carsFile, parserLib = parserLib, inferSchema = true)
```
Comment (falaki) [inline src/test/scala/com/databricks/spark/csv/CsvSuite.scala]: Just wondering, why did you change this to use ```csvFile()```?

Comment (brkyvz) [inline src/test/scala/com/databricks/spark/csv/CsvSuite.scala]: I just copied csvFastSuite here :)

**Comment by falaki** [comment]
LGTM. Merging it now.
